sklearn/manifold/tests/test_t_sne.py

```diff
@@ -1198,6 +1198,57 @@ def test_tsne_n_jobs(method):
     assert_allclose(X_tr_ref, X_tr)
 
 
+# TODO: Remove filterwarnings in 1.2
+@pytest.mark.filterwarnings("ignore:.*TSNE will change.*:FutureWarning")
+def test_tsne_with_mahalanobis_distance():
+    """Make sure that method_parameters works with mahalanobis distance"""
+    random_state = check_random_state(0)
+    n_features = 10
```

Comment:
Since we are using the same parameters everywhere, we can refactor this test:

```python
@pytest.mark.filterwarnings("ignore:.*TSNE will change.*:FutureWarning")
def test_tsne_with_mahalanobis_distance():
    """Make sure that method_parameters works with mahalanobis distance."""
    random_state = check_random_state(0)
    n_features = 10
    n_samples = 300
    X = random_state.randn(n_samples, n_features)
    default_params = {
        "perplexity": 40,
        "n_iter": 250,
        "learning_rate": "auto",
        "n_components": 3,
        "random_state": 0,
    }

    # 1. raises error here (original issue)
    tsne = TSNE(metric="mahalanobis", **default_params)
    ref = "Must provide either V or VI for Mahalanobis distance"
    with pytest.raises(ValueError, match=ref):
        tsne.fit_transform(X)

    # 2. check for correct answer
    precomputed_X = squareform(pdist(X, metric="mahalanobis"), checks=True)
    ref = TSNE(metric="precomputed", **default_params).fit_transform(precomputed_X)

    now = TSNE(
        metric="mahalanobis", metric_params={"V": np.cov(X.T)}, **default_params
    ).fit_transform(X)
    assert_allclose(ref, now)
```

Also note the `assert_allclose` instead of the `assert_array_equal`.